app/src/main/java/com/example/tasky/feature_login/domain/repository/UserRemoteRepository.kt

```diff
@@ -0,0 +1,11 @@
+package com.example.tasky.feature_login.domain.repository
+
+import com.example.tasky.common.model.Resource
+import com.example.tasky.feature_login.domain.model.LoginUserInfo
+import com.example.tasky.feature_login.domain.model.LoginUserResponse
+import com.example.tasky.feature_login.domain.model.RegisterUserInfo
+
+interface UserRemoteRepository {
+    suspend fun postRegisterCall(registerUserInfo: RegisterUserInfo): Resource<Unit>
+    suspend fun postLoginCall(loginUserInfo: LoginUserInfo): Resource<LoginUserResponse>
```

Comment:
Strictly speaking, the names of these functions also contain implementation details (how the request to the API needs to look like) which domain doesn't need to know. You want to abstract all implementation details out to the data layer to only expose what's really needed (`"Hey, this function is for registering a user, all I need is an email and a password"`)